core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -327,27 +339,87 @@ private String getConfigurationFileName() {
         ClassLoader classLoader = TelemetryConfigurationFactory.class.getClassLoader();
         URL resource = classLoader.getResource(fileToParse);
 
-        // If not found as a resource, trying to load from the executing jar directory
         if (resource == null) {
-            try {
-                String jarFullPath = TelemetryConfigurationFactory.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
-                File jarFile = new File(jarFullPath);
-
-                if (jarFile.exists()) {
-                    String jarDirectory = jarFile.getParent();
-                    String configurationPath = jarDirectory + File.separator + fileToParse;
+            InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.INFO, "Did not find resource configuration file");
 
-                    InternalLogger.INSTANCE.trace("Found configuration file: '%s'", configurationPath);
-                    return configurationPath;
-                }
-            } catch (URISyntaxException e) {
-                InternalLogger.INSTANCE.error("Failed to find configuration file: '%s'", e.getMessage());
+            // If not found as a resource, trying to load from the executing jar directory
+            String configurationFile = getConfFromSourceLocation();
+            if (configurationFile != null) {
+                return configurationFile;
             }
```

Comment:
It may be a religious thing, but generally I prefer having as little `else`s as possible. The reason for that is that once we complete with the `if` part we either returned or continuing in the main flow of the function and not in a sub-flow.
This also makes the code less nested, especially in cases where there are several nested conditions.